kimchi/src/lagrange_basis_evaluations.rs

```diff
@@ -98,7 +108,38 @@ impl<F: FftField> LagrangeBasisEvaluations<F> {
 
         // Denominators now contains the desired result.
         LagrangeBasisEvaluations {
-            evals: denominators,
+            evals: vec![denominators],
+        }
+    }
+
+    /// Compute all evaluations of the normalized lagrange basis polynomials of the
+    /// given domain at the given point. Runs in time O(domain size).
+    fn new_with_chunked_segments(
+        max_poly_size: usize,
+        domain: D<F>,
+        x: F,
+    ) -> LagrangeBasisEvaluations<F> {
+        let n = domain.size();
+        let num_chunks = n / max_poly_size;
+        let mut evals = Vec::with_capacity(num_chunks);
+        for i in 0..num_chunks {
+            let mut x_pow = F::one();
+            let mut chunked_evals = vec![F::zero(); n];
+            for j in 0..max_poly_size {
+                chunked_evals[i * max_poly_size + j] = x_pow;
+                x_pow *= x;
+            }
+            domain.ifft_in_place(&mut chunked_evals);
```

Comment:
An inverse fast-Fourier transform. Identical reasoning to [here](https://github.com/o1-labs/proof-systems/blob/769362887694142e2eb46dbcd1ce305902ddffba/poly-commitment/src/srs.rs#L87), except that we're using powers of ζ instead of the commitments.

Reply:
Updated the comment.